src/ucp/core/ucp_worker.c

```diff
@@ -2524,28 +2527,39 @@ static void ucp_worker_destroy_eps(ucp_worker_h worker,
     }
 }
 
+static void ucp_worker_eps_cleanup(ucp_worker_h worker)
+{
+    /* Cleanup hash of discarded UCT EPs prior destroying all UCP EPs to
+     * destroy UCP EPs which were fully discarded */
+    ucp_worker_discard_uct_ep_cleanup(worker);
+    ucp_worker_destroy_eps(worker, &worker->all_eps, "all");
+    ucp_worker_destroy_eps(worker, &worker->internal_eps, "internal");
+    /* Cleanup hash of discarded UCT EPs one more time to destroy UCP EPs
+     * for which UCT EP discarding procedure were started during destroying
+     * UCP EPs */
```

Comment:
how it that possible?

Reply:
it could be the discarding of AUX EPs of WIRUEP EPs.
added an explanation